bootloader/src/old_machine_bootloader_windows.c

```diff
@@ -217,6 +173,10 @@ int sendRequest(wchar_t* server, wchar_t* tunnel, BOOL tunnelUsed, wchar_t* reqD
     BOOL isSend = HttpSendRequest(hRequest, NULL, 0, reqData, (DWORD)(sizeof(wchar_t) * wcslen(reqData)));
     if (!isSend){
         printf("HttpSendRequest error : (%lu)\n", GetLastError());
+        // close request
+        InternetCloseHandle(hRequest);
+        InternetCloseHandle(hConnect);
+        InternetCloseHandle(hInternet);
```

Comment:
The correct pattern is to initialize them all to invalid values and in the end of the function create a label of cleanup.
in the cleanup, check each handle against the known invalid value and if different, close it.
then you have a bunch of goto cleanup statements

Reply:
Note that otherwise, you leak `hInternet` if `InternetConnect` fails. That's why goto cleanup is a good idiom